app/src/ui/repository.tsx

```diff
@@ -209,7 +208,7 @@ export class RepositoryView extends React.Component<
         : null) || null
 
     // -1 Because of right hand side border
-    const availableWidth = clamp(this.props.sidebarWidth) - 1
+    const availableWidth = this.props.sidebarWidth.value - 1
```

Comment:
This doesn't seem like the correct approach. The thought behind clamping here is that we attempt to preserve the user's preferred width as the window changes. So if they have set the sidebar width to 300px and have the horizontal real estate to accommodate that but then resize the app to the smallest width the sidebar will get compressed to its minimum width only for it to return to 300px again when the window is resized.

If we need to make the sidebar smaller when zoomed then that logic should get incorporated into the `updateResizableConstraints` method in AppStore.

Note that the same clamping operation happens for the repo list foldout as well.

Lastly, but only semi-related to this PR, we should probably change the constrain logic to match how CSS gives precedence to min-width.

```diff
diff --git a/app/src/lib/stores/app-store.ts b/app/src/lib/stores/app-store.ts
index a8702e62a..95eeec32c 100644
--- a/app/src/lib/stores/app-store.ts
+++ b/app/src/lib/stores/app-store.ts
@@ -7759,5 +7764,8 @@ function constrain(
   min = -Infinity,
   max = Infinity
 ): IConstrainedValue {
+  // Match CSS's behavior where min-width takes precedence over max-width
+  // See https://stackoverflow.com/a/16063871
+  max = max < min ? min : max
   return { value: typeof value === 'number' ? value : value.value, min, max }
 }
```

Reply:
> This doesn't seem like the correct approach. The thought behind clamping here is that we attempt to preserve the user's preferred width as the window changes. So if they have set the sidebar width to 300px and have the horizontal real estate to accommodate that but then resize the app to the smallest width the sidebar will get compressed to its minimum width only for it to return to 300px again when the window is resized.

Thanks for that explanation.. I was trying to figure out why it was there since I didn't see a change in behavior.. but I didn't not thinking of that use case.

Reply:
I think your matching of css behavior will solve this actually because the constrain is applied in the `updateResizableConstraints `. Applying and testing now.

Reply:
Yep. Works great!